safedelete/queryset.py

```diff
@@ -27,13 +49,44 @@ def delete(self, force_policy=None):
             :py:func:`safedelete.models.SafeDeleteModel.delete`
         """
         assert self.query.can_filter(), "Cannot use 'limit' or 'offset' with delete."
-        # TODO: Replace this by bulk update if we can
-        for obj in self.all():
-            obj.delete(force_policy=force_policy)
-        self._result_cache = None
+
+        with transaction.atomic():
+            all_objs = self.all()
+            all_objs_id_list = list(all_objs.values_list('id', flat=True))
+            current_policy = force_policy or all_objs.model._safedelete_policy
+
+            if current_policy == HARD_DELETE_NOCASCADE:
```

Comment:
We should also split the logic of each branch in their own method as it was done in https://github.com/makinacorpus/django-safedelete/commit/800458629a4ec48c51b17388a79e3f95e6f0a770, in order to be able to override them.